queries/qm/query_mods.go

```diff
@@ -20,14 +33,39 @@ func (m queryMods) Apply(q *queries.Query) {
 // Apply the query mods to the Query object
 func Apply(q *queries.Query, mods ...QueryMod) {
 	for _, mod := range mods {
-		mod(q)
+		mod.Mod(q)
 	}
 }
 
+type sqlQueryMod struct {
+	sql  string
+	args []interface{}
+}
+
+// Mod implements QueryMod.Mod.
+func (qm *sqlQueryMod) Mod(q *queries.Query) {
+	queries.SetSQL(q, qm.sql, qm.args...)
+}
+
 // SQL allows you to execute a plain SQL statement
 func SQL(sql string, args ...interface{}) QueryMod {
-	return func(q *queries.Query) {
-		queries.SetSQL(q, sql, args...)
+	return &sqlQueryMod{
+		sql:  sql,
+		args: args,
+	}
+}
+
+type loadQueryMod struct {
+	relationship string
+	mods         []QueryMod
+}
+
+// Mod implements QueryMod.Mod.
+func (qm *loadQueryMod) Mod(q *queries.Query) {
```

Comment:
These intermediate structs are very small and are always read-only after creation so there's no reason to use pointers to them, this will just generate excess garbage I think. Though because we're using interfaces I think we end up with a pointer to them anyway so it might not save us anything in reality. But we should at least try to be proper. Let's change these all to be non-pointer receivers. (Talking about the `loadQueryMod` struct here).

Reply:
Done.